src/main/kotlin/org/elm/lang/core/types/TypeInference.kt

```diff
@@ -436,6 +436,7 @@ private class InferenceScope(
             atom is ElmLetInExpr -> inferChild { beginLetInInference(atom) }.ty
             atom is ElmCaseOfExpr -> inferCase(atom)
             atom is ElmParenthesizedExpr -> inferExpression(atom.expression)
+            atom is ElmRecordExpr -> inferRecord(atom)
```

Comment:
Allowing Psi errors when inferring records worries me a little, but I haven't found any problems with it yet. By making this more relaxed, we may start to see weird type inference results caused by the Psi tree being messed up. And it may be difficult to debug.

That being said, if this feature works out, then I'd be ok with the tradeoff.